File: tutorial/patterns/p17.cpp
#include <iostream>
using namespace std;

int main(){
    /*
                    *
                *   *
            *   *   * 
        *   *   *   * 
    
    */

    int n ;
    cin >> n;
    int i = 1;

    while (i<=n)
    {
        int j = 1;
        int space = n-i;
        while(space){
            cout << " " << " ";
            space--;
            }
        while(j <= i){
            cout << "*" << " ";
            j++;
        }
        cout << endl;        
        i++;
    }
    
}           

File: tutorial/patterns/p17.dart
import 'dart:io';

void main() {
  /*
                    *
                *   *
            *   *   * 
        *   *   *   * 
    
    */

  int n = int.parse(stdin.readLineSync()!);
  int i = 1;
  while (i <= n) {
    int j = 1;
    int space = n - i;
    while (j <= space) {
      stdout.write(" ");
      space--;
    }
    while (j <= i) {
      stdout.write("*");
      j++;
    }
    stdout.writeln();
    i++;
  }
}
